python-aliases: cleanup and keep sorted

qweered opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

qweered committed:
python-aliases: remove inherit

qweered committed:
python-aliases: cleanup and add missing comments

Co-authored-by: Wolfgang Walther <walther@technowledgy.de>

wolfgangwalther commented on pkgs/top-level/python-aliases.nix:
```diff
@@ -286,7 +282,6 @@ mapAliases {
   postgrest-py = throw "'postgrest-py' has been renamed to/replaced by 'postgrest'" postgrest; # Converted to throw 2025-10-28
   powerlineMemSegment = throw "'powerlineMemSegment' has been renamed to/replaced by 'powerline-mem-segment'" powerline-mem-segment; # Converted to throw 2025-10-28
   prayer-times-calculator = throw "'prayer-times-calculator' has been renamed to/replaced by 'prayer-times-calculator-offline'" prayer-times-calculator-offline; # Converted to throw 2025-10-28
-  premailer = throw "premailer was removed, as it is not compatible with lxml>4.9.4.";
```
Needs to have a date added.

wolfgangwalther commented on pkgs/top-level/python-aliases.nix:
```diff
@@ -274,7 +271,6 @@ mapAliases {
   orderedset = throw "orderedset has been removed because it has been marked as broken since at least November 2024."; # Added 2025-10-04
   orm = throw "orm has been removed as it is unmaintained"; # added 2025-08-27
   paho-mqtt_2 = throw "'paho-mqtt_2' has been renamed to/replaced by 'paho-mqtt'" paho-mqtt; # Converted to throw 2025-10-28
-  pam = python-pam; # added 2020-09-07.
```
Probably just needs to remove the `.`

wolfgangwalther commented on pkgs/top-level/python-aliases.nix:
```diff
@@ -356,10 +351,8 @@ mapAliases {
   python-lz4 = throw "'python-lz4' has been renamed to/replaced by 'lz4'" lz4; # Converted to throw 2025-10-28
   python_magic = throw "'python_magic' has been renamed to/replaced by 'python-magic'" python-magic; # Converted to throw 2025-10-28
   python_mimeparse = throw "'python_mimeparse' has been renamed to/replaced by 'python-mimeparse'" python-mimeparse; # Converted to throw 2025-10-28
-  python-Levenshtein = levenshtein;
```
Not sure why this is flat out remove. Shouldn't this be turned into a throw?

wolfgangwalther commented on pkgs/top-level/python-aliases.nix:
```diff
@@ -356,10 +351,8 @@ mapAliases {
-  python-Levenshtein = levenshtein;
   python-simple-hipchat = throw "'python-simple-hipchat' has been removed because it was broken and unmaintained"; # added 2025-08-26
   python-subunit = throw "'python-subunit' has been renamed to/replaced by 'subunit'" subunit; # Converted to throw 2025-10-28
-  python-wifi = throw "python-wifi does not support Python3";
```
Let's add a date before automation kicks in.

wolfgangwalther reviewed (dismissed):
Only looked at `python-aliases: remove alisases not catched by script` this time.

wolfgangwalther commented on pkgs/top-level/python-aliases.nix:
```diff
@@ -177,7 +178,7 @@ mapAliases {
   criticality-score = throw "use pkgs.criticality-score instead"; # added 2024-12-31
   cryptacular = throw "cryptacular was removed, because it was disabled on all python version since 3.6 and last updated in 2021"; # Added 2024-05-13
-  cryptography_vectors = "cryptography_vectors is no longer exposed in python*Packages because it is used for testing cryptography only."; # Added 2022-03-23
+  cryptography_vectors = throw "'cryptography_vectors' has been renamed to/replaced by 'cryptography-vectors'" cryptography-vectors; # Converted to throw 2025-10-28
```
This is wrong. Ths adds a second argument to `throw`, which is `cryptography-vectors`. We don't need that, only need the original string.

qweered committed:
python-aliases: drop throws older than june 2025

qweered committed:
python-aliases: convert to throw aliases before june 2025

qweered committed:
python-aliases: keep sorted

wolfgangwalther commented:
First few commits LGTM at first glance now, I will checkout the branch locally now and work through it to see whether I can still find anything else :)

wolfgangwalther committed:
python-aliases: revert some throws not present in 25.05

These were dated to before the release, but were actually merged later,
so are not present in 25.05.

wolfgangwalther reviewed (approved):
Thank you very much for all the cleanup work, appreciated! :)
